### PR TITLE
`pallet-staking`: Expose missing consts

### PR DESCRIPTION
**Pull request by bkchr**

### CODE REVIEW

File: frame/staking/src/lib.rs [hunk 1]
```diff
@@ -1275,6 +1275,32 @@ decl_module! {
...
+		/// The number of blocks before the end of the era from which election submissions are allowed.
+		/// Setting this to zero will disable the offchain compute and only on-chain seq-phragmen will
+		/// be used.
+		///
+		/// This is bounded by being within the last session. Hence, setting it to a value more than the
+		/// length of a session will be pointless.
+		const ElectionLookahead: T::BlockNumber = T::ElectionLookahead::get();
+
+		/// Maximum number of balancing iterations to run in the offchain submission. If set to 0,
+		/// balance_solution will not be executed at all.
```
Comment (joepetrowski) [inline frame/staking/src/lib.rs:1292]:
```suggestion
		/// `balance_solution` will not be executed at all.
```

File: frame/staking/src/lib.rs [hunk 2]
```diff
@@ -1275,6 +1275,32 @@ decl_module! {
...
+		const MaxIterations: u32 = T::MaxIterations::get();
+
+		/// The threshold of improvement that should be provided for a new solution to be accepted.
+		const MinSolutionScoreBump: Perbill = T::MinSolutionScoreBump::get();
+
+		/// The maximum number of nominator rewarded for each validator.
```
Comment (joepetrowski) [inline frame/staking/src/lib.rs:1298]:
```suggestion
		/// The maximum number of nominators rewarded for each validator.
```

File: frame/staking/src/lib.rs [hunk 3]
```diff
@@ -1275,6 +1275,32 @@ decl_module! {
 		/// Number of eras that staked funds must remain bonded for.
 		const BondingDuration: EraIndex = T::BondingDuration::get();
 
+		/// Number of eras that slashes are deferred by, after computation. This should be less than the
+		/// bonding duration. Set to 0 if slashes should be applied immediately, without opportunity for
+		/// intervention.
```
Comment (NikVolf) [inline frame/staking/src/lib.rs:1280]:
```suggestion
		/// Number of eras that slashes are deferred by, after computation. 
		///
		/// This should be less than the bonding duration. 
		/// Set to 0 if slashes should be applied immediately, without opportunity for
		/// intervention.
```

File: frame/staking/src/lib.rs [hunk 4]
```diff
@@ -1275,6 +1275,32 @@ decl_module! {
...
+		const SlashDeferDuration: EraIndex = T::SlashDeferDuration::get();
+
+		/// The number of blocks before the end of the era from which election submissions are allowed.
+		/// Setting this to zero will disable the offchain compute and only on-chain seq-phragmen will
+		/// be used.
+		///
+		/// This is bounded by being within the last session. Hence, setting it to a value more than the
+		/// length of a session will be pointless.
```
Comment (NikVolf) [inline frame/staking/src/lib.rs:1288]:
```suggestion
		/// The number of blocks before the end of the era from which election submissions are allowed.
		///
		/// Setting this to zero will disable the offchain compute and only on-chain seq-phragmen will
		/// be used.
		///
		/// This is bounded by being within the last session. Hence, setting it to a value more than the
		/// length of a session will be pointless.
```

File: frame/staking/src/lib.rs (hunk 1 above)
Comment (NikVolf) [inline frame/staking/src/lib.rs:1292]:
```suggestion
		/// Maximum number of balancing iterations to run in the offchain submission. 
		///
		/// If set to 0, balance_solution will not be executed at all.
```

**Comment by NikVolf** [comment]
Title in docs should be separated by empty `///` for better html output ^^^

**Comment by kianenigma** [comment]
bot merge

**Comment by bkchr** [comment]
bot merge

**Comment by bkchr** [comment]
> Head SHA has changed since merge was requested; aborting merge.

intelligent bot :D

**Comment by sjeohp** [comment]
bot merge

**Pull request merged**